agent-control/src/version_checker/handler.rs

```diff
@@ -0,0 +1,56 @@
+use crate::agent_control::agent_id::AgentID;
+use crate::sub_agent::error::SubAgentError;
+use crate::version_checker::AgentVersion;
+use opamp_client::StartedClient;
+use opamp_client::opamp::proto::{AnyValue, KeyValue, any_value};
+
+/// This method request the AgentDescription from the current opamp client and, updates or add the
+/// field from agent version to be sent to opamp server
+pub fn on_version<C>(
+    agent_data: AgentVersion,
+    agent_id: &AgentID,
+    maybe_opamp_client: Option<&C>,
+) -> Result<(), SubAgentError>
+where
+    C: StartedClient,
+{
+    if let Some(client) = maybe_opamp_client.as_ref() {
+        let mut agent_description = client.get_agent_description()?;
+        match agent_id {
+            AgentID::AgentControl | AgentID::SubAgent(_) => {
+                agent_description.identifying_attributes =
+                    update_version_key_values(agent_description.identifying_attributes, agent_data);
+            }
+            AgentID::K8sCD => {
+                // TODO CHANGEME after discussion with FC
+                agent_description.non_identifying_attributes = update_version_key_values(
+                    agent_description.non_identifying_attributes,
+                    agent_data,
+                )
+            }
```

Comment:
We are also using the new agent_id here 🤔. If we finally removed it we would need to do this differently as well. Maybe we can decouple both the handler and the version checker from the AgentID variants.

Reply:
Reworked it by exposing the desired `opamp_field`. LMKWYT!